orders/convoy.go

```diff
@@ -202,6 +202,40 @@ func ConvoyDestinations(v godip.Validator, src godip.Province, noConvoy *godip.P
 	return result
 }
 
+func ConvoySources(v godip.Validator, dst godip.Province, noConvoy *godip.Province) (result []godip.Province) {
```

Comment:
Same thing here - please avoid duplicating long and complicated function like this - send the difference as a flag or parameter instead?